openedx/core/djangoapps/cors_csrf/middleware.py

```diff
@@ -145,3 +128,120 @@ def process_response(self, request, response):
+                # protection. This is true for any HTTP connection, but anyone
+                # using HTTPS expects better! For this reason, for
+                # https://example.com/ we need additional protection that treats
+                # http://example.com/ as completely untrusted. Under HTTPS,
+                # Barth et al. found that the Referer header is missing for
+                # same-domain requests in only about 0.2% of cases or less, so
+                # we can use strict Referer checking.
+                referer = force_text(
+                    request.META.get('HTTP_REFERER'),
+                    strings_only=True,
+                    errors='replace'
+                )
+                if referer is None:
+                    return self._reject(request, django_csrf.REASON_NO_REFERER)
+
+                # Here we generate a list of all acceptable HTTP referers,
+                # including the current host since that has been validated
+                # upstream.
+                good_hosts = list(settings.CSRF_TRUSTED_ORIGINS)
+                # Note that request.get_host() includes the port.
+                good_hosts.append(request.get_host())
+                good_referers = ['https://{0}/'.format(host) for host in good_hosts]
+                if not any(same_origin(referer, host) for host in good_referers):
+                    reason = REASON_BAD_REFERER % referer
+                    return self._reject(request, reason)
+
+            if csrf_token is None:
+                # No CSRF cookie. For POST requests, we insist on a CSRF cookie,
+                # and in this way we can avoid all CSRF attacks, including login
+                # CSRF.
+                return self._reject(request, django_csrf.REASON_NO_CSRF_COOKIE)
+
+            # Check non-cookie token for match.
+            request_csrf_token = ""
+            if request.method == "POST":
+                try:
+                    request_csrf_token = request.POST.get('csrfmiddlewaretoken', '')
+                except IOError:
+                    # Handle a broken connection before we've completed reading
+                    # the POST data. process_view shouldn't raise any
+                    # exceptions, so we'll ignore and serve the user a 403
+                    # (assuming they're still listening, which they probably
+                    # aren't because of the error).
+                    pass
+
+            if request_csrf_token == "":
+                # Fall back to X-CSRFToken, to make things easier for AJAX,
+                # and possible for PUT/DELETE.
+                request_csrf_token = request.META.get('HTTP_X_CSRFTOKEN', '')
+
+            if not constant_time_compare(request_csrf_token, csrf_token):
+                return self._reject(request, django_csrf.REASON_BAD_TOKEN)
+
+        return self._accept(request)
+
+
+class CorsCSRFMiddleware(CrossDomainCsrfViewMiddleware):
```

Comment:
This middleware is not needed. AFAIK, Django already supports CORS for subdomains through the  `CSRF_TRUSTED_ORIGINS` settings alongside `CSRF_COOKIE_DOMAIN = '.example.com'`.

I have no idea why would this be needed.

Please check: https://docs.djangoproject.com/en/1.11/ref/csrf/ for more info.

cc: @Fahminajjar @Salomari1987 and @devalih

Reply:
Moved to https://github.com/Edraak/edraak-platform/pull/92